src/components/Button/Button.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import { darken } from 'polished';
+
+const colorStyles = css`
+  ${({ theme, color }) => {
+    const selected = theme.colors[color];
+    return css`
+      background: ${selected};
+      color: ${selected === '#FFF' ? 'black' : 'white'};
+      &:hover {
+        background: ${darken(0.1, selected)};
+      }
+      &:active {
+        background: ${darken(0.1, selected)};
+      }
+      ${props =>
+        props.outline &&
+        css`
+          color: ${selected};
+          background: white;
+          border: 1px solid ${selected};
+          &:hover {
+            background: ${selected};
+            color: white;
+          }
+        `}
+    `;
+  }}
+`;
+
+const fontWeightStyles = css`
+  ${({ fontWeight }) => {
+    const selected = fontWeight;
+    return css`
+      font-weight: ${selected};
+    `;
+  }}
+`;
+
+const sizes = {
+  large: {
+    height: '3rem',
+    fontSize: '1.25rem',
+  },
+  medium: {
+    height: '2.25rem',
+    fontSize: '1rem',
+  },
+  small: {
+    height: '1.75rem',
+    fontSize: '0.875rem',
+  },
+};
+
+const sizeStyles = css`
+  ${({ size }) => css`
+    height: ${sizes[size].height};
+    font-size: ${sizes[size].fontSize};
+  `}
+`;
+
+const fullWidthStyle = css`
+  ${props =>
+    props.fullWidth &&
+    css`
+      width: 100%;
+      justify-content: center;
+    `}
+`;
+
+const StyledButton = styled.button`
+  ${({ theme }) => theme.flexSet()};
+  outline: none;
+  border: 1px solid rgb(230, 230, 230);
+  border-radius: 4px;
+  color: black;
+  cursor: pointer;
+  padding-left: 1rem;
+  padding-right: 1rem;
+  margin-bottom: 0.6rem;
+
+  ${fontWeightStyles}
+
+  ${sizeStyles}
+
+  ${colorStyles}
+
+  ${fullWidthStyle}
+`;
+
+function Button({
+  children,
+  color,
+  size,
+  fontWeight,
+  fullWidth,
+  outline,
+  ...rest
+}) {
+  return (
+    <StyledButton
+      color={color}
+      size={size}
+      outline={outline}
+      fullWidth={fullWidth}
+      fontWeight={fontWeight}
+      {...rest}
+    >
+      {children}
+    </StyledButton>
+  );
+}
+
+Button.defaultProps = {
+  color: 'white',
+  size: 'medium',
+  fontWeight: '400',
+};
```

Comment:
- 파일 열었을 때 다른 내용들 보다 컴포넌트를 먼저 확인할 수 있도록 순서를 변경하면 어떨까요?
- defalutProps를 사용할 거라면 전부 rest props로 넘겨줘도 될거 같고, 그렇게 안할거라면 기본 파라미터 값 사용해서 넘겨주면 될거 같아요.
```suggestion
function Button({
  children,
  color = 'white',
  size = 'medium',
  fontWeight = '400',
  fullWidth,
  outline,
  ...rest
}) {
  return (
    <StyledButton
      color={color}
      size={size}
      outline={outline}
      fullWidth={fullWidth}
      fontWeight={fontWeight}
      {...rest}
    >
      {children}
    </StyledButton>
  );
}
```

Reply:
넵! 컴포넌트를 앞쪽으로 쓰겠습니다!
만약에 전부 rest props로 전달한다면, 
```
 &lt;StyledButton   {...rest}    &gt;
```
이렇게 되겠죠//?